code/game/gamemodes/shadowling/shadowling_abilities.dm

```diff
@@ -135,6 +135,9 @@
 			extinguishMob(H)
 		for(var/mob/living/silicon/robot/borg in T.contents)
 			borg.update_headlamp(1, charge_max)
+		for(var/mob/living/silicon/pai/B in T.contents)
```

Comment:
This handles pAIs on their own on turfs, but does it handle pAIs in a user's pocket/on their head/in their hand?

Reply:
I'm going to beat you up if you keep suggesting ways this can go terribly wrong >:^(

Reply:
@Spacemanspark I'd advocate adding a check for a held pAI in `extinguishMob`, which would take care of that case